GeoLib/QuadTree.h

```diff
@@ -381,16 +384,20 @@ template <typename POINT> class QuadTree
 		POINT mid_point(_ll);
 		mid_point[0] += (_ur[0] - _ll[0]) / 2.0;
 		mid_point[1] += (_ur[1] - _ll[1]) / 2.0;
+		assert(_childs[0] == nullptr);
```

Comment:
The assertions were quite useful to track down (my) programming errors. I would not gather them together at the beginning of the method, because I want to keep the assertions as near as possible at the location the condition should be valid.